tests/experimental/test_experimental_base_simulator.py

```diff
@@ -0,0 +1,198 @@
+        min_val, max_val = mock_simulator._parameters_range[param_name]
+        assert torch.all(samples[:, i] >= min_val)
+        assert torch.all(samples[:, i] <= max_val)
+
+
+def test_forward(mock_simulator):
+    """Test that forward method works correctly"""
+    # Create test input
+    test_input = torch.tensor([[0.5, 0.0, 7.5]], dtype=torch.float32)
+
+    # Get output
+    output = mock_simulator.forward(test_input)
+
+    # Check shape
+    assert output.shape == (1, len(mock_simulator._output_names))
+
+    # Check values against expected transformation
+    expected_sum = 0.5 + 0.0 + 7.5
+    assert output[0, 0] == pytest.approx(expected_sum * 1)
+    assert output[0, 1] == pytest.approx(expected_sum * 2)
+
+
+def test_forward_batch(mock_simulator):
+    """Test that forward_batch method processes multiple samples correctly"""
+    # Create test batch
+    n_samples = 3
+    batch = torch.tensor(
+        [[0.5, 0.0, 7.5], [0.2, 0.3, 6.0], [0.8, -0.5, 9.0]], dtype=torch.float32
+    )
+
+    # Process batch
+    results = mock_simulator.forward_batch(batch)
+
+    # Check shape
+    assert results.shape == (n_samples, len(mock_simulator._output_names))
+
+    # Check values for each sample
+    for i in range(n_samples):
+        expected_sum = sum(batch[i].tolist())
+        assert results[i, 0] == pytest.approx(expected_sum * 1)
+        assert results[i, 1] == pytest.approx(expected_sum * 2)
+
+
+def test_get_parameter_idx(mock_simulator):
+    """Test that get_parameter_idx returns correct indices"""
+    assert mock_simulator.get_parameter_idx("param1") == 0
+    assert mock_simulator.get_parameter_idx("param2") == 1
+    assert mock_simulator.get_parameter_idx("param3") == 2
+
+    # Test invalid parameter
+    with pytest.raises(ValueError, match="Parameter .* not found"):
+        mock_simulator.get_parameter_idx("invalid_param")
+
+
+def test_properties(mock_simulator, parameters_range, output_names):
+    """Test that properties return correct values"""
+    # Test param_names
+    assert mock_simulator.param_names == list(parameters_range.keys())
+
+    # Test param_bounds
+    assert mock_simulator.param_bounds == list(parameters_range.values())
+
+    # Test output_names
+    assert mock_simulator.output_names == ["var1", "var2"]
+
+    # Test output_names
+    assert mock_simulator.output_names == output_names
+
+
+def test_abstract_class():
+    """Test that Simulator cannot be instantiated directly"""
+    params = {"param": (0, 1)}
+    outputs = ["out"]
+
+    # Should raise TypeError when trying to instantiate abstract base class
+    with pytest.raises(TypeError):
+        Simulator(params, outputs)  # type: ignore[abstract]
+
+
+def test_handle_simulation_failure():
+    """Test handling of simulation failures in forward_batch"""
+
+    # This test demonstrates that the current implementation doesn't properly handle
+    # None returns (which would be a good enhancement)
```

Comment:
Looking at this I was unsure where the `None` handling is demonstrated / tested in this test?

Reply:
It's more there as a reminder (not a test) that at the moment we don't handle the simulator failing. The NaghaviSimulator currently returns None but in this set up we don't allow for that. It's something to be addressed as part of #438